tests/amd_migraphx_test/runMIGraphXTests/runMIGraphXTests.cpp

```diff
@@ -0,0 +1,676 @@
+using namespace cv;
+
+#if USE_OPENCV_4
+#define CV_LOAD_IMAGE_COLOR IMREAD_COLOR
+#endif
+
+#define ERROR_CHECK_STATUS(status) { \
+    vx_status status_ = (status); \
+    if (status_ != VX_SUCCESS) { \
+        printf("ERROR: failed with status = (%d) at " __FILE__ "#%d\n", status_, __LINE__); \
+        exit(1); \
+    } \
+}
+
+#define ERROR_CHECK_OBJECT(obj) { \
+    vx_status status_ = vxGetStatus((vx_reference)(obj)); \
+    if(status_ != VX_SUCCESS) { \
+        printf("ERROR: failed with status = (%d) at " __FILE__ "#%d\n", status_, __LINE__); \
+        exit(1); \
+    } \
+}
+
+static void VX_CALLBACK log_callback(vx_context context, vx_reference ref, vx_status status, const vx_char string[])
+{
+    size_t len = strnlen(string, MAX_STRING_LENGTH);
+    if (len > 0) {
+        printf("%s", string);
+        if (string[len - 1] != '\n')
+            printf("\n");
+        fflush(stdout);
+    }
+}
+
+inline int64_t clockCounter()
+{
+    return std::chrono::high_resolution_clock::now().time_since_epoch().count();
+}
+
+inline int64_t clockFrequency()
+{
+    return std::chrono::high_resolution_clock::period::den / std::chrono::high_resolution_clock::period::num;
+}
+
+void show_usage() {
+    printf(
+            "\n"
+            "Usage: ./runmigraphxtests \n"
+            "--imagenet_image          <path to image from imagenet dataset>\n"
+            "--mnist_image             <path to image from mnist dataset>\n"
+            "--profiler_mode  <range:0-2; default:0> [optional]\n"
+            "  Mode 0 - Run all tests\n"
+            "  Mode 1 - Run all ONNX tests\n"
+            "  Mode 2 - Run all JSON tests\n"
+            "--profiler_level <range:0-N; default:1> [N = batch size][optional]\n"
+            "--mnist          <mnist-model>     \n"
+            "--resnet50       <resnet50-model>  \n"
+            "--googlenet      <googlenet-model> \n" 
+            "--squeezenet     <resnet101-model> \n"
+            "--alexnet        <resnet152-model> \n"
+            "--vgg19          <vgg19-model>     \n"
+            "--densenet       <densenet-model>  \n"
+            "\n"
+        ); 
+}
+
+int main(int argc, char **argv) {
+
+    // check command-line usage
+    std::string binaryFilename_squeezenet_str;
+    std::string binaryFilename_resnet50_str;
+    std::string binaryFilename_vgg19_str;
+    std::string binaryFilename_googlenet_str;
+    std::string binaryFilename_alexnet_str;
+    std::string binaryFilename_densenet_str;
+    std::string binaryFilename_mnist_str;
+    std::string imagenet_inputFile_str;
+    std::string mnist_inputFile_str;
+
+    int parameter = 0;
+    int64_t freq = clockFrequency(), t0, t1;
+    int N = 1000;
+
+    for(int arg = 1; arg < argc; arg++) {
+        if (!strcasecmp(argv[arg], "--help") || !strcasecmp(argv[arg], "--H") || !strcasecmp(argv[arg], "--h")) {
+            show_usage();
+            exit(-1);
+        }
+        else if (!strcasecmp(argv[arg], "--imagenet_image")) {
```

Comment:
@LakshmiKumar23 as Discussed please make changes so that the test runs individual models as well.

Reply:
@asalmanp Made the changes. User can choose which models to run from list. 
Also fixed codacy warnings, unless something new pops up